apisix/plugins/kafka-logger.lua

```diff
@@ -153,10 +143,29 @@ function _M.log(conf, ctx)
         return
     end
 
-    local api_ctx = {}
-    api_ctx.conf_type = ctx.conf_type
-    api_ctx.conf_id = ctx.conf_id
-    api_ctx.conf_version = ctx.conf_version
+    -- reuse producer via lrucache to avoid unbalanced partitions of messages in kafka
+    local broker_list = {}
```

Comment:
The `broker_list` can be allocated with `core.table.new`, to pre-allocate the table.